include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -133,6 +133,10 @@ typedef enum OrtErrorCode {
   ORT_EP_FAIL,
 } OrtErrorCode;
 
+typedef enum OrtSessionConfigKey {
+  ORT_SESSION_CONFIG_MAXIMUM,  // Always leave this as the last value
+} OrtSessionConfigKey;
```

Comment:
Should entries in this list have explicit values?

Reply:
Add a min value with explicit value, 

---
In reply to: [466696899](https://github.com/microsoft/onnxruntime/pull/4718#discussion_r466696899) [](ancestors = 466696899)

Reply:
From what I've seen users have a tendency to just use an integer instead of the enum value if they can get away with it which potentially breaks if they update the ORT version and something was inserted in the middle (causing us to have to respond to a github issue). Due to that it's way safer to use explicit values for all entries.

But that makes me wonder if we really want to use an enum for the key instead of a static list of known config names. The enum will get disorganised very quickly as entries get added whereas a list of valid config names can have a much better organized structure as things can be added in appropriate places.

e.g. say I had ["ep.cuda.use_arena", "arena.bfc.initial_size"] and I wanted to add "ep.openvino.use_arena" I can add that in the middle so all the 'ep' values are together. if we have an enum with explicitly numbered entries for stability I have to either add it at the end or have out-of-order entries in the enum.

---
In reply to: [466752374](https://github.com/microsoft/onnxruntime/pull/4718#discussion_r466752374) [](ancestors = 466752374,466696899)